config/config.go

```diff
@@ -2,50 +2,73 @@ package config
 
 import (
 	"crypto/tls"
+	"encoding/json"
 	"fmt"
 	"log"
-	"net/url"
 	"os"
 	"path/filepath"
 	"runtime"
-	"strings"
 
 	"github.com/Scalingo/envconfig"
 	"github.com/Scalingo/go-scalingo"
 	"github.com/stvp/rollbar"
+	"gopkg.in/errgo.v1"
 )
 
+type ConfigFile struct {
+	Region string `json:"region"`
+}
+
 type Config struct {
-	ScalingoApiUrl       string
-	apiHost              string
-	token                string
 	ApiVersion           string
 	DisableInteractive   bool
 	DisableUpdateChecker bool
-	SshHost              string
 	UnsecureSsl          bool
 	RollbarToken         string
-	ConfigDir            string
-	AuthFile             string
-	LogFile              string
-	logFile              *os.File
-	Logger               *log.Logger
+
+	// Override region configuration
+	ScalingoApiUrl  string
+	ScalingoAuthUrl string
+	ScalingoDbUrl   string
+	ScalingoRegion  string
+	ScalingoSshHost string
+
+	// Configuration files
+	ConfigDir      string
+	AuthFile       string
+	LogFile        string
+	ConfigFilePath string
+	ConfigFile     ConfigFile
+
+	// Cache related files
+	CacheDir         string
+	RegionsCachePath string
+
+	// Logging
+	logFile *os.File
+	Logger  *log.Logger
 }
 
 var (
-	AuthenticatedUser *scalingo.User
-	env               = map[string]string{
-		"SCALINGO_API_URL": "https://api.scalingo.com",
-		"SSH_HOST":         "scalingo.com:22",
-		"API_VERSION":      "1",
-		"UNSECURE_SSL":     "false",
-		"ROLLBAR_TOKEN":    "",
-		"CONFIG_DIR":       ".config/scalingo",
-		"AUTH_FILE":        "auth",
-		"LOG_FILE":         "local.log",
-	}
-	C         Config
-	TlsConfig *tls.Config
+	env = map[string]string{
+		"SCALINGO_AUTH_URL":  "https://auth.scalingo.com",
+		"SCALINGO_API_URL":   "",
+		"SCALINGO_DB_URL":    "",
+		"SCALINGO_SSH_HOST":  "scalingo.com:22",
+		"SCALINGO_REGION":    "",
+		"API_VERSION":        "1",
+		"UNSECURE_SSL":       "false",
+		"ROLLBAR_TOKEN":      "",
+		"CONFIG_DIR":         ".config/scalingo",
```

Comment:
Will it work properly on Windows or MacOS? Shouldn't we use each OS standard? Maybe through a lib such as https://github.com/shibukawa/configdir#configdir-for-golang?

Same question for the cache

Reply:
Actually I'm using exactly the same thing as we're doing for the authentication file which is also in .config/scalingo

There is some custom code for Windows/MacOS lower in that file

Reply:
That being said we're not exactly using the same PATH historically, but it would require a migration, I'm in favor of doing it in another issue, I'm opening it

Reply:
https://github.com/Scalingo/cli/issues/426